### PR TITLE
Fix/2830 enter many names

### LINKED ISSUE #2830: Immediately resolves name as soon as I enter ".eth", but this excludes the ability to enter many names
**Describe the bug**
When entering an ENS name into a Recipient field, as soon as I enter a possible ENS suffix, e.g. ".eth" or ".xyz", the app immediately resolves the name and doesn't let me keep typing. This is a problem because subdomains exist in ENS. E.g. the name evan.ethereum.eth exists, but when I try to type it in as soon as I get to "evan.eth..." it resolves the name and doesn't let me keep typing. This would also be a problem if I tried to send to "brantly.xyz.eth", the app cuts me off as soon as i typed in "brantly.xyz", even though "brantly.xyz.eth" and "brantly.xyz" are two different names with different owners and records.

**Screenshots**
_If applicable, add screenshots or links to help explain your problem_

**To Reproduce**
_Steps to reproduce the behavior_
1. Go to MetaMask Mobile on Android
2. Tap "Send"
3. Try to type in to the Recipient field the ENS name "evan.ethereum.eth"
4. You'll see that it cuts you off as soon as you type in "evan.eth..."

**Expected behavior**
You should be able to type in any name, and the app should never stop you from typing in more. As you type in, could present possible names below for you to select, but it shouldn't stop you from typing more.

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy S21+
 - OS: latest Android
 - App Version: v2.4.0 (746)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
Could lead to lost funds

- Add stats if available on % of customers impacted
potentially everyone

- Is this visible to all users?
yes

- Is this tech debt?
no, you just have a wrong implementation of ENS

### PR DESCRIPTION
**Pull request by tommasini**
**Description**

The input text is not longer transformed in a formatted text with was impossible to keep writing.
Now it is a text input with a check icon that tell us that the address it is valid.

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #2830 
Resolves the fact that we could not write many names like "evan.ethereum.ether" and now we can access that addresses too with this solution.

Small demo with the final result

https://user-images.githubusercontent.com/46944231/160585472-64d17744-04c8-47ed-93d6-d1ed27c8c660.mp4

### CODE REVIEW

**Comment by chrisleewilcox** [comment]
When an addressed is resolved, should we not show the warning to confirm it's a valid address? Since the address is resolved?  Should the warning only show for addresses and names not resolved?

**Review by chrisleewilcox — CHANGES_REQUESTED** [review]
Fixed detox addressbook-tests.spec.js that broke with this branch.

**Review by chrisleewilcox — CHANGES_REQUESTED** [review]
Fixed detox addressbook-tests.spec.js that broke with this branch.

**Review by chrisleewilcox — APPROVED** [review]
LGTM

File: app/util/address/index.js [hunk 1]
```diff
@@ -50,10 +50,10 @@ export function renderShortAddress(address, chars = 4) {
 	return `${checksummedAddress.substr(0, chars + 2)}...${checksummedAddress.substr(-chars)}`;
 }
 
-export function renderSlightlyLongAddress(address, chars = 4) {
+export function renderSlightlyLongAddress(address, chars = 4, initialChars = 20) {
 	if (!address) return address;
 	const checksummedAddress = toChecksumAddress(address);
-	return `${checksummedAddress.substr(0, chars + 20)}...${checksummedAddress.substr(-chars)}`;
+	return `${checksummedAddress.substr(0, chars + initialChars)}...${checksummedAddress.substr(-chars)}`;
```
Comment (gantunesr) [inline app/util/address/index.js:56]: Can you replace this deprecated method?

File: app/util/address/index.test.ts [hunk 2]
```diff
@@ -17,3 +17,21 @@ describe('isENS', () => {
 		expect(isENS('ricky.metamask.eth')).toBe(true);
 	});
 });
+
+describe('renderSlightlyLongAddress', () => {
+	it('should return the address when the address do not exist', () => {
+		expect(renderSlightlyLongAddress(null)).toBe(null);
+	});
+	it('should return 5 characters before ellipsis and 4 final characters of the address after the ellipsis', () => {
+		expect(renderSlightlyLongAddress('0xC4955C0d639D99699Bfd7Ec54d9FaFEe40e4D272').split('.')[0].length).toBe(24);
```
Comment (gantunesr) [inline app/util/address/index.test.ts:26]: You could move `'0xC4955C0d639D99699Bfd7Ec54d9FaFEe40e4D272'` to a variable?

```
const mockAddress = '0xC4955C0d639D99699Bfd7Ec54d9FaFEe40e4D272'
```

**Review by gantunesr — CHANGES_REQUESTED** [review]
Good work man, just a few comments

- Check why ENS is showing instead of address
- Can you verify that this flow is working well with ERC721 (NFTs)

File: app/util/address/index.js (hunk 1 above)
Comment (tommasini) [reply app/util/address/index.js]: Replaced for the slice method

File: app/util/address/index.test.ts (hunk 2 above)
Comment (tommasini) [reply app/util/address/index.test.ts]: Done it! Thank you!

**Review by chrisleewilcox — APPROVED** [review]
Retested and tested confirm screen with screenshots provided.

LGTM

**Review by gantunesr — APPROVED** [review]
LGTM

**Pull request merged**